### PR TITLE
Cleans up some machinery harddels

### PR DESCRIPTION
**Pull request by carlarctg**
# About the pull request

Cleans up some machinery harddels.

# Explain why it's good for the game

Hard dels are when an object is destroyed without having all its references (variables that are set to track another atom/datum, such as var/x = new(src) ). These are very expensive to delete due to BYOND moments, reducing performance.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
code: Cleans up some machinery harddels
/:cl:

### CODE REVIEW

File: code/game/machinery/bio-dome_floodlights.dm [hunk 1]
```diff
@@ -19,6 +19,13 @@
 		F.fswitch = src
 	start_processing()
 
+/obj/structure/machinery/hydro_floodlight_switch/Destroy()
+	for(var/obj/structure/machinery/hydro_floodlight/floodlight as anything in floodlist)
+		QDEL_NULL(floodlight.fswitch)
+	QDEL_NULL_LIST(floodlist)
+	..()
```
Comment (Zonespace27) [inline code/game/machinery/bio-dome_floodlights.dm:26]: return parent

File: code/game/machinery/bio-dome_floodlights.dm [hunk 2]
```diff
@@ -19,6 +19,13 @@
 		F.fswitch = src
 	start_processing()
 
+/obj/structure/machinery/hydro_floodlight_switch/Destroy()
+	for(var/obj/structure/machinery/hydro_floodlight/floodlight as anything in floodlist)
```
Comment (Zonespace27) [inline code/game/machinery/bio-dome_floodlights.dm:23]: would QDEL NULL LIST work here

File: code/game/machinery/groundmap_geothermal.dm [hunk 3]
```diff
@@ -225,6 +225,13 @@
 		F.fswitch = src
 	start_processing()
 
+/obj/structure/machinery/colony_floodlight_switch/Destroy()
+	for(var/obj/structure/machinery/colony_floodlight/floodlight as anything in floodlist)
```
Comment (Zonespace27) [inline code/game/machinery/groundmap_geothermal.dm:229]: ditto

File: code/game/machinery/groundmap_geothermal.dm [hunk 4]
```diff
@@ -225,6 +225,13 @@
...
+		QDEL_NULL(floodlight.fswitch)
+	QDEL_NULL_LIST(floodlist)
+	..()
```
Comment (Zonespace27) [inline code/game/machinery/groundmap_geothermal.dm:232]: return

**Review by Zonespace27 — DISMISSED** [review]

File: code/game/machinery/bio-dome_floodlights.dm (hunk 2 above)
Comment (carlarctg) [reply code/game/machinery/bio-dome_floodlights.dm:23]: i don't think so - you can't do 

`QDEL_NULL_LIST(floodlist.fswitch)` or whatever

File: code/game/machinery/groundmap_geothermal.dm (hunk 3 above)
Comment (carlarctg) [reply code/game/machinery/groundmap_geothermal.dm:229]: as above

**Pull request merged**